conanfile.py

```diff
@@ -0,0 +1,132 @@
+from conans import ConanFile, CMake, tools
+import platform
+import shutil
+import urllib.request
+import tarfile
+import sys
+import os
+
+class TaraxaConan(ConanFile):
+    name = "taraxa-node"
+    version = "0.1"
+    description = "Taraxa is a Practical Byzantine Fault Tolerance blockchain."
+    topics = ("blockchain", "taraxa", "crypto")
+    url = "https://github.com/Taraxa-project/taraxa-node"
+    homepage = "https://www.taraxa.io"
+    license = "MIT"
+
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def _add_clang_utils_on_darwin(self):
+        clang_format = "clang-format"
+        clang_tidy = "clang-tidy"
+        path_to_format = shutil.which(clang_format)
+        path_to_tidy = shutil.which(clang_tidy)
+        find_format = shutil.which(clang_format) != None or os.path.exists(clang_format)
+        find_tidy = shutil.which(clang_tidy) != None or os.path.exists(clang_tidy)
+        if not find_format or not find_tidy:
+            print("downloading LLVM...")
+            dirname = "clang+llvm-10.0.0-x86_64-apple-darwin"
+            llvm_bin_link = "https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/" + dirname + ".tar.xz"
+            ftpstream = urllib.request.urlopen(llvm_bin_link)
+            thetarfile = tarfile.open(fileobj=ftpstream, mode="r|xz")
+            thetarfile.extractall()
+            if not find_format:
+                shutil.move(dirname + "/bin/" + clang_format, clang_format)
+                find_format = True
+            if not find_tidy:
+                shutil.move(dirname + "/bin/" + clang_tidy, clang_tidy)
+                find_tidy = True
+            shutil.rmtree(dirname)
+        if find_format and path_to_format == None:
+            path_to_format = os.path.abspath(os.getcwd()) + "/" + clang_format
+        if find_tidy and path_to_tidy == None:
+            path_to_tidy = os.path.abspath(os.getcwd()) + "/" + clang_tidy
+        print("clang-format path: " + path_to_format)
+        print("clang-tidy path: " + path_to_tidy)
+
+    def requirements(self):
+        self.requires("boost/1.71.0")
+        self.requires("cppcheck/2.3")
+        self.requires("openssl/1.1.1f")
+        self.requires("cryptopp/8.4.0")
+        self.requires("gtest/1.10.0")
+        self.requires("rocksdb/6.8.1")
+        self.requires("jsoncpp/1.9.4")
+        self.requires("gmp/6.2.0")
+        self.requires("mpfr/4.0.2")
+        self.requires("snappy/1.1.8")
+        self.requires("zstd/1.4.4")
+        self.requires("lz4/1.9.2")
+
```

Comment:
Just thinking out loud now. Maybe we could have a separate file with lib versions for this script. Might be cleaner but maybe unnecessary - don't know :)

Reply:
This file is intended to list everything on the project, not in the style of conan to write multiple files =)